pkg/apis/application/v1alpha1/types.go

```diff
@@ -49,6 +49,8 @@ type ApplicationSpec struct {
 	IgnoreDifferences []ResourceIgnoreDifferences `json:"ignoreDifferences,omitempty" protobuf:"bytes,5,name=ignoreDifferences"`
 	// Infos contains a list of useful information (URLs, email addresses, and plain text) that relates to the application
 	Info []Info `json:"info,omitempty" protobuf:"bytes,6,name=info"`
+	// Always use hard-refresh for application refreshing
+	AlwaysHardRefresh bool `json:"alwaysHardRefresh,omitempty" protobuf:"bytes,7,opt,name=alwaysHardRefresh"`
```

Comment:
Can we please have always hard refresh in a separate PR?

This feature looks pretty straightforward but we need to be more careful about it otherwise it will affect performance. I'm trying to list edge cases I will post it to https://github.com/argoproj/argo-cd/pull/1794